src/hooks/api/getters/useEvents/useEvents.ts

```diff
@@ -0,0 +1,52 @@
+import useSWR, { mutate, SWRConfiguration } from 'swr';
+import { useCallback } from 'react';
+import { formatApiPath } from '../../../../utils/format-path';
+import handleErrorResponses from '../httpErrorResponseHandler';
+import { IEvent } from '../../../../interfaces/event';
+
+const PATH = formatApiPath('api/admin/events');
+
+export interface UseEventsOutput {
+    events: IEvent[];
+    refetchEvents: () => void;
+    loading: boolean;
+    error?: Error;
+}
+
+export const useEvents = (
+    featureName?: string,
+    options?: SWRConfiguration
+): UseEventsOutput => {
+    const { data, error } = useSWR<{ events: IEvent[] }>(
+        [PATH, featureName],
+        () => fetchEvents(featureName),
+        options
+    );
+
+    const refetchEvents = useCallback(() => {
+        mutate(PATH).catch(console.warn);
+    }, []);
+
+    return {
+        events: data?.events || [],
+        loading: !error && !data,
+        refetchEvents,
+        error,
+    };
+};
+
+function fetchEvents(featureName?: string) {
```

Comment:
We've tried to use arrow functions as much as possible for new code, just to keep consistency and eventually reach a state where things are similar enough that it reduces the cognitive load.